### PR TITLE
fix: [CDS-50616]: Do not pass pipeline branch name to template request when gitEntityInfo is null

### PR DESCRIPTION
**Pull request by vcalasansh**
## Describe your changes
Before this change, the `branch` query parameter that was set in `GitSyncBranchContext` was being propagated to a call to Template Service's `applyTemplatesOnGivenYamlV2`, even in the case where `GitContextHelper.getGitEntityInfo()` returned `null` (in which case, the call to template service should be made without passing any GitSync context params).

This was causing error in Trigger's create/update endpoints when the underlying pipeline has a reference to a remote template that exists in a different branch of the pipeline itself.

These changes fix this behavior by wrapping the call to template service in a `GitSyncBranchContextGuard` with empty branch context.

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/42576)

### CODE REVIEW

**Review by sribalij — DISMISSED** [review]
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42576)* status: 1 file(s) to review. 1 of 2 approvals obtained.

**Review by adivishy1 — APPROVED** [review]
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42576)* status: :shipit: complete! ✓

**Review by mohitg0795 — APPROVED** [review]
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42576)* status: 1 of 2 approvals obtained.

**Review by rafaelmart1ns — APPROVED** [review]
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42576)* status: :shipit: complete! ✓

File: pipeline-service/service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineTemplateHelper.java [hunk 1]
```diff
@@ -92,14 +94,17 @@ public TemplateMergeResponseDTO resolveTemplateRefsInPipeline(String accountId,
...
-            TemplateApplyRequestDTO.builder()
-                .originalEntityYaml(yaml)
-                .checkForAccess(checkForTemplateAccess)
-                .getMergedYamlWithTemplateField(getMergedTemplateWithTemplateReferences)
-                .build()));
-
+        GitSyncBranchContext gitSyncBranchContext =
+            GitSyncBranchContext.builder().gitBranchInfo(GitEntityInfo.builder().build()).build();
+        try (PmsGitSyncBranchContextGuard ignored = new PmsGitSyncBranchContextGuard(gitSyncBranchContext, true)) {
+          return NGRestUtils.getResponse(templateResourceClient.applyTemplatesOnGivenYamlV2(accountId, orgId, projectId,
```
Comment (vcalasansh) [inline pipeline-service/service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineTemplateHelper.java:100]: this client call

**Pull request merged**